packages/flutter/lib/src/material/ink_sparkle.dart

```diff
@@ -617,7 +518,8 @@ class FragmentShaderManager {
 
   /// Compiles the spir-v bytecode into a shader program.
   Future<void> compile() async {
-    _program = await ui.FragmentProgram.compile(spirv: spirvByteBuffer);
+    final ByteData data = await rootBundle.load('shaders/ink_sparkle.frag');
+    _program = await ui.FragmentProgram.compile(spirv: data.buffer);
```

Comment:
This still isn't safe but we're no worse off than we used to be. We should file an issue to deprecate this and remove it. It's not safe for anyone to use.

Reply:
@dnfield Can you be more specific about what isn't safe, and what should be removed?

Reply:
Sorry, providing sync access to a late initialized variable that gets initialized in a future.

So if some program invokes the getter below here that refers to this static late var before the future completes, it will get a late initialization exception and it has no way to know if that getter is safe to call yet or not.

Reply:
I think that depends on whether the size of the blobs means that they need to be read from the asset bundle asynchronously. I'm not sure how big e.g. the msl blobs will be but the spir-v for the sparkle shader is already ~11KB.

Reply:
I guess technically we should be doing that async anyway to avoid jank on low end devices

Reply:
@dnfield What is the unsafe code pass? The only way to obtain an instance of this is to await `inkSparkle` "static factory" above. At that point, it is guaranteed that _program is initialized, no?

(Side-node: I think `compile` should be a private method since it should only be invoked by `inkSparkle`?)

Reply:
(we could assert that _program is initialized in the inkSparkle method before the return)

Reply:
Ahhhh I see, in that case this is fine. I missed that part because of the way GitHub collapsed part of the diff...